### DISCUSSION TITLE
Commit status do not allow emojis

### DISCUSSION THREAD

**Discussion by samuelcolvin** [community/community#29034] [Apps, API and Webhooks]
Either via gh markdown, or from unicode.

Submitted: `description='Change file looks good 🎉'`

Response:

```
GithubException: 422 
{"message": "Validation Failed", "errors": [{"resource": "Status", "code": "custom", "field": "description", 
"message": "description doesn't accept 4-byte Unicode"}], 
"documentation_url": "https://docs.github.com/rest/commits/statuses#create-a-commit-status"}
```

**Comment by pasindu651** [comment]


**Reply by samuelcolvin** [reply]
I'm not talking about `git commit`, I'm talking about the API endpoint. This is clear from my issue description.

> Either via gh markdown

As I stated in my issue description, I've already tried `:tada:`.

**Comment by GuillermoFarias** [comment]
here you have available the list of emojis accepted by the api https://api.github.com/emojis
On this repo have a visual representation of that api https://github.com/ikatyang/emoji-cheat-sheet
`:rocket:` => 🚀

**Reply by samuelcolvin** [reply]
Please read my issue!

This is not what I'm talking about.

**Comment by larrytheliquid** [comment]
Bumping this thread. I agree with @samuelcolvin, while I can appreciate that the GH status API might not support 4-byte unicode directly, a nice workaround would be able to replace unicode like 🎉 with `:tada:`, and then have the GitHub status render the 🎉 emoji in the GH status UI!

**Comment by mikem-of** [comment]
Running into the same issue... I want to render an emoji in my checks, using the GitHub API. This seems not possible. 

From the [docs](https://docs.github.com/en/rest/commits/statuses?apiVersion=2022-11-28#create-a-commit-status) it's not clear that you can't POST an emoji as part of the context or description. 

_Note: I CAN use emojis in the name field in a Github Actions workflow. That renders an emoji in the check section. So it should be technically possible._ 

### 1. This works
```
gh api \
  --method POST \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  /repos/OWNER/REPO/statuses/SHA \
  -f 'description=Succes' -f 'context=Build'
```

Renders this check:
<img width="246" height="56" alt="image" src="https://github.com/user-attachments/assets/0b027faa-226e-4858-bbe0-2216da99e833" />


### 2. Adding an emoji to either description or context returns an error
```
gh api \
  --method POST \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  /repos/OWNER/REPO/statuses/SHA \
  -f 'description='Succes 🎉' -f 'context=Build'
```

Error:
```
{
  "message": "Validation Failed",
  "errors": "Validation failed: Description doesn't accept 4-byte Unicode",
  "documentation_url": "https://docs.github.com/rest/commits/statuses#create-a-commit-status",
  "status": "422"
}
```

### 3. Replacing the emoji with the representation as described [here](https://api.github.com/emojis) does not error, but does not render an emoji
```
gh api \
  --method POST \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2022-11-28" \
  /repos/OWNER/REPO/statuses/SHA \
  -f 'description='Succes :tada:' -f 'context=Build'
```

Renders this check:
<img width="281" height="51" alt="image" src="https://github.com/user-attachments/assets/c4cbc7c2-42d5-40bf-b10d-ee2f8f2ba8e7" />


Any help would be much appreciated!
